### PR TITLE
#2448: re-style error states for Donor Summary Table

### PR DESCRIPTION
**Pull request by samrichca**
# Description of changes

- add errors for missing matched pairs, samples, and raw reads
- add errors to legend

## Type of Change

- [ ] Bug
- [ ] Dependency updates
- [x] Feature
- [ ] Refactoring
- [ ] Release candidate
- [x] Styling
- [ ] Testing
- [ ] Other (please describe)

## Checklist before requesting review

- Design (select one):
  - [x] Matches design:
    - component sizes, spacing, and styles
    - font size, weight, colour
    - spelling has been double checked
  - [ ] No design provided, screenshot of changes included in PR
  - [ ] No changes to UI
- Back-end (select one):
  - [ ] Back-end PRs merged, tested on develop, and linked in this PR
  - [x] No back-end changes
- [x] Manually tested changes
- [x] Added copyrights to new files
- [x] Connected ticket to PR

### CODE REVIEW

File: components/pages/submission-system/common.tsx [hunk 1]
```diff
@@ -118,11 +126,26 @@ export const TableLegendSection = styled('div')`
   margin-bottom: 10px;
 `;
 
-export const TableLegendEntry = styled('div')`
-  margin-right: 5px;
-  display: flex;
-  align-items: flex-start;
-`;
+export const TableLegendEntry = ({
```
Comment (samrichca) [inline components/pages/submission-system/common.tsx:129]: I had to tweak the styling for this, so I refactored it to cut down on some code repetition too.

File: components/pages/submission-system/program-dashboard/DonorDataSummary/DesignationCell.tsx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+export const DesignationCellLegacy = ({ left, right }: { left: number; right: number }) => {
+  const theme = useTheme();
+  const isValid = (num: number) => num > 0;
+
+  const DesignationEntry = styled('div')`
+    text-align: center;
+    flex: 1;
+    color: ${(props: { num: number }) =>
+      isValid(props.num) ? theme.colors.primary : theme.colors.error};
+  `;
```
Comment (ciaranschutte) [inline components/pages/submission-system/program-dashboard/DonorDataSummary/DesignationCell.tsx:112]: this `styled` div can be moved outside component. `theme` should be accessible from the styled function call

**Review by ciaranschutte — APPROVED** [review]
can move a styled component out of regular component. looks great!

File: components/pages/submission-system/program-dashboard/DonorDataSummary/DesignationCell.tsx (hunk 2 above)
Comment (samrichca) [reply components/pages/submission-system/program-dashboard/DonorDataSummary/DesignationCell.tsx]: added, it made me add a theme prop but 🤷‍♀️

**Pull request merged**